lib/api/v3/queries/query_representer.rb

```diff
@@ -217,6 +217,30 @@ class QueryRepresenter < ::API::Decorators::Single
                     end
                   }
 
+        resources :highlighted_attributes,
+                  getter: ->(*) {
+                    represented.highlighted_columns.map do |column|
+                      ::API::V3::Queries::Columns::QueryColumnsFactory.create(column)
+                    end
+                  },
+                  setter: ->(fragment:, **) {
+                    columns = Array(fragment).map do |column|
+                      name = id_from_href "queries/columns", column['href']
+
+                      ::API::Utilities::PropertyNameConverter.to_ar_name(name, context: WorkPackage.new) if name
+                    end
+
+                    represented.highlighted_attributes = columns.map(&:to_sym).compact if fragment
```

Comment:
The setter is not tested by the current tests:
https://github.com/opf/openproject/blob/dev/spec/lib/api/v3/queries/query_representer_parsing_spec.rb

Reply:
Done